platforms/posix/src/main.cpp

```diff
@@ -366,6 +368,9 @@ int main(int argc, char **argv)
 				cout << "node name: " << node_name << endl;
 			}
 
+		} else if (strchr(argv[index], '>')) {
```

Comment:
What actually encountered in Debian on BBBlue was that the output redirect was passed to argv.

Reply:
Weird. What are the individual `argv` strings and how do you run it? Does it actually do the redirect?

Reply:
Checked my porting notes. When run from command line, argv does not contain the redirect part. The above redirect handling was added when handling issues in px4 auto start during boot. This also brings up another issue encountered when setting up auto start on BeagleBone Blue. Here are some of my porting notes:

Notes for autostart using systemd:
( px4-*.service currently generated too much log and  increased disk usage quickly
  so use /etc/rc.local for now
  BTW, both methods are working except the above px4 problem
)

Scripts and comments in my /etc/rc.local:

#/home/debian/px4/px4  /home/debian/px4/posix-configs/px4.config > /home/debian/px4/PX4.log &
#
# After initial output, it continuosly fill up the log with the following garbage data:
#     pxh> ESC[2K^Mpxh> <FF>ESC[2K^Mpxh> <FF><FF>ESC[2K^Mpxh> <FF><FF><FF>ESC[2K^Mpxh> <FF><FF><FF><FF>ESC[2K^Mpxh> ......
# Disk usage can jump from about 50% to 80% quickly
# so have to redirect output to /dev/null for now
#
/home/debian/px4/px4  /home/debian/px4/posix-configs/px4.config  > /dev/null &

Reply:
You need to start it with: `/home/debian/px4/px4 -d /home/debian/px4/posix-configs/px4.config > /home/debian/px4/PX4.log &`

> The above redirect handling was added when handling issues in px4 auto start during boot.

Does `strcmp` work as well? I just want to avoid introducing corner cases where the parsing is incorrect. Can you please double-check that it's still an issue?

Reply:
"-d" optiion worked and I removed the redirect flag and handled the case when test_data is absent. Previously when I used "-d /home/debian/px4" option, a link like the following was created, and caused failure when re-launching px4:

debian@beaglebone:~/px4$ ls -l
lrwxrwxrwx 1 root   root         26 Jul 31 19:02 test_data -> /home/debian/px4/test_data

I mistaken the whole of "-d /home/debian/px4" as data directory option and always avoided using the "-d" later.

The test_data directory is not small ( > 1MB )  and I do not think it's necessary to run px4, so I changed code not to create this link when the target directory does not exist.